.github/workflows/reusable-docker-build-publish.yml

```diff
@@ -288,6 +302,32 @@ jobs:
             exit 1
           fi
 
+      - name: Check docker cache constraints
+        id: should-cache
+        env:
+          GITHUB_REF_NAME: ${{ inputs.github-ref-name }}
+          GITHUB_EVENT_NAME: ${{ inputs.github-event-name }}
+          GITHUB_TRUNK_BRANCH: ${{ inputs.github-trunk-branch }}
```

Comment:
We could get this from the `gh` CLI in case you think that's better

```
❯ gh repo view --json defaultBranchRef --jq .defaultBranchRef.name
main
```

Reply:
That seems like a better solution than adding another required input. Will update and retest.

Reply:
retested: https://github.com/smartcontractkit/chainlink/actions/runs/15448537732/job/43484708882?pr=17978